src/clj/game/cards/hardware.clj

```diff
@@ -195,9 +200,9 @@
                  :choices {:req #(and (is-type? % "Program")
                                       (= (:zone %) [:discard]))}
                  :effect (req (when (>= (:credit runner) (:cost target))
-                                    (do (runner-install state side target)
-                                        (trash state side card {:cause :ability-cost})
-                                        (system-msg state side (str "uses " (:title card) " to install " (:title target))))))}]}
+                                (do (runner-install state side target)
```

Comment:
Do you need this `do` here?

Reply:
Nope, but I was trying to avoid any "functional" changes. I'm fine removing these, lol.